src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java

```diff
@@ -209,6 +240,88 @@ private WorkflowStatus checkWorkflowStatus(String workflowId, Long workspaceId)
         }
     }
 
+    private List<Long> obtainLabelIDs(@Nullable Long workspaceId) throws ApiException {
+
+        if (labels == null || labels.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        if (!labelPermission(workspaceId)) {
+            throw new ApiException("User does not have permission to modify pipeline labels");
+        }
+
+        // retrieve labels for the workspace and check if we need to create new ones
+        List<LabelDbDto> wspLabels = new ArrayList<>();
+
+        ListLabelsResponse res = api().listLabels(workspaceId, null, null, null, LabelType.SIMPLE, null);
+        if (res.getLabels() != null) {
+            wspLabels.addAll(res.getLabels());
+        }
+
+        Map<String, Long> nameToID = wspLabels
+            .stream()
+            .collect(Collectors.toMap(LabelDbDto::getName, LabelDbDto::getId));
+
+        // get label names not registered in workspace (names are unique per wspID)
+        List<String> newLabels = labels
+            .stream()
+            .filter(labelName -> !nameToID.containsKey(labelName))
+            .collect(Collectors.toList());
+
+        // create the new ones via POST /labels
+        for (String labelName: newLabels) {
+            CreateLabelResponse created = api().createLabel(
+                new CreateLabelRequest()
+                    .name(labelName)
+                    .resource(false)
+                    .isDefault(false),
+                workspaceId
+            );
+            nameToID.put(created.getName(), created.getId());
+        }
+
+        // map requested label names to label IDs
+        return labels
+            .stream()
+            .map(nameToID::get)
+            .collect(Collectors.toList());
+    }
+
+    private boolean labelPermission(@Nullable Long wspId) throws ApiException {
+
+        // personal workspace
+        if (wspId == null) return true;
+
+        var client = HttpClient.newBuilder()
+            .connectTimeout(Duration.ofSeconds(10))
+            .build();
+
+        var uri = UriBuilder
+            .fromUri(URI.create(apiUrl() + "/permissions"))
```

Comment:
Shouldn't this be part of the SDK? The rest of the CLI is always using the Tower SDK to call Tower API.

Reply:
Is something blocking to add the `/permissions` endpoint to the SDK? But nevermind, as you wish, but this is the typical thing that theoretically it's temporal but it ends up here forever. So at least add a big TODO comment here explaining why this was done like this and when it should be removed.